prov/gni/src/gnix_vc.c

```diff
@@ -1199,6 +1199,43 @@ int _gnix_vc_destroy(struct gnix_vc *vc)
 	}
 
 	/*
+	 * move vc state to terminating
+	 */
+
+	vc->conn_state = GNIX_VC_CONN_TERMINATING;
+
+	/*
+	 * try to unbind the gni_ep if non-NULL.
+	 * If there are SMSG or PostFMA/RDMA outstanding
+	 * wait here for them to complete
+	 */
+
+	if (vc->gni_ep != NULL) {
+		while (status == GNI_RC_NOT_DONE) {
+
+			fastlock_acquire(&nic->lock);
+			status = GNI_EpUnbind(vc->gni_ep);
+			fastlock_release(&nic->lock);
+
+			if ((status != GNI_RC_NOT_DONE) &&
+				(status != GNI_RC_SUCCESS))
+				GNIX_WARN(FI_LOG_EP_CTRL,
+					"GNI_EpUnBind returned %s\n",
+					  gni_err_str[status]);
```

Comment:
I think we should just break out of loop in case of error, since it probably won't correct itself.